src/main/java/com/amihaiemil/camel/YamlSerializer.java

```diff
@@ -0,0 +1,39 @@
+package com.amihaiemil.camel;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * A Yaml serializer.
+ * @author sherif
+ * @version $Id$
+ * @since 1.0.0
+ */
+public interface YamlSerializer {
+    
+    /**
+     * Serialize a Map<Object, Object> to YamlNode.
+     * @param map Map<Object, Object>
+     * @return Yaml node
+     * @TODO #30 Implement the serialize method with Map input and unit-test it.
```

Comment:
@SherifWaly The format of the puzzle is almost correct, but not entirely. The letters ``todo`` need to be lowercase, and after ``#30``, you need to add the estimation and role: ``:30m/DEV``. Also, the text needs to be changed. One good puzzle would be:

```
@todo #30:30m/DEV This interface should have 3 implementor classes: YamlObjectSerializer,
 YamlMapSerializer and YamlCollectionSerializer. Each of these classes will encapsulate and serialize
 the mentioned types (i.e. Object, Map<Object, Object> and Collection<Object). The return type
 or method ``serialize()`` should be overridden with the proper subtype (e.g. YamlObjectSerializer will
 have the method YamlMapping serialize() {...}
```

Notice that at the beginning of each new line, there is a space (it is important because otherwise, the bot won't parse it right). ``#30`` is the parent ticket, while ``:30m/DEV`` is the estimation and role.

Now, of course, the puzzle asks for a lot of effort, which will take more than 30min, for sure, but this puzzle will be broken in more puzzles. Whoever will take it, will implement just one part of it (one of the classes), and will leave other puzzles in the code to cover the rest of the task. Makes sense? :)